packages/js/src/decorator/gutenberg.js

```diff
@@ -311,6 +311,22 @@ export function calculateAnnotationsForTextFormat( text, mark ) {
 	return blockOffsets;
 }
 
+/**
+ * Create an annotation if the given mark is position based.
+ *
+ * @param {Mark}   mark The mark to apply to the content.
+ *
+ * @returns {Array} The annotations to apply.
+ */
+export function crateAnnotationsFromPositionBasedMarks( mark ) {
+	return [
+		{
+			startOffset: mark.getBlockPositionStart(),
+			endOffset: mark.getBlockPositionEnd(),
+		},
+	];
+}
```

Comment:
Do I understand correctly that in order to get the information of the start and end position of the block, we need to pass them through `position.startOffsetBlock` / `position.endOffsetBlock` when we create the `Mark` object inside each assessment?
If yes, then how do we calculate them?

Reply:
Also, what happens if we have multiple matches with different positions in the text? Shouldn't we create multiple annotations then in that case? 🤔

Reply:
NOTE: with the current approach, `crateAnnotationsFromPositionBasedMarks` helper will create one annotation from one Mark object that is applicable for the current block/annotatable attribute of the block.

So if there are multiple matches or multiple Mark objects that are applicable, then we'll also create the annotations for them.

Reply:
Update from Aida: It's already the case (with the current approach) to create multiple annotations when there are multiple matches. But crateAnnotationsFromPositionBasedMarks helper is design to return one annotation each time.